### DISCUSSION TITLE
03Tools - Pipeline Example 1 - missing a step

### DISCUSSION THREAD

**Discussion by pattansa** [statmike/vertex-ai-mlops#12] [Q&A] [answered]
Thank you for all the content that you have provided. It is really nice to go through it and learn. I think there is a small step missing so wanted to point out.

I am getting an error running the pipeline_job.wait()
"Failed to create pipeline job. Error: Service account `1081626357791-compute@developer.gserviceaccount.com` does not have `[storage.objects.get, storage.objects.create]` IAM permission(s) to the bucket

I think it is missing the step of copying the compile package to GCS bucket like we are doing in Example 2.

**Comment by pattansa** [comment]
Nevermind, it was just a permission issue. Once I added the permission of service account to the GCS bucket, the pipeline is running. 
I might have missed an earlier step of giving permission to the bucket.

**Comment by statmike** [comment, marked as answer]
Hello @pattansa,
Thank you for pointing this out.  I did have a commented out note in the Setup section where the service account is retrieved about needing roles/storage.objectAdmin permissions.  Today I pushed an update that adds a short section to the [00 - Environment Setup] to help with this one time permissions setup.  All the notebooks that use the SERVICE_ACCOUNT now have a check and a note that points back to the setup.  Let me know if this would have helped with the issue you ran into.
